chainer/training/extensions/linear_shift.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import division
+
+from chainer.training import extension
+
+
+class LinearShift(extension.Extension):
+
+    """Trainer extension to change an optimizer attribute linearly.
+
+    This extension changes an optimizer attribute from the first value to the
+    last value linearly within a specified duration. The typical use case is
+    warming up of the momentum coefficient.
+
+    For example, suppose that this extension is called at every iteration, and
+    ``value_range == (x, y)`` and ``time_range == (i, j)``. Then, this
+    extension keeps the attribute to be ``x`` up to the ``i``-th iteration, and
+    linearly shifts the value to ``y`` by the ``j``-th iteration.
```

Comment:
As for the second comment, it is actually described at the head of this paragraph.
For the third comment, it is slightly complicated: `_t` is 1 at the first iteration as you noted, and this is exactly matching to the iteration count of the optimizer, because at the first iteration the optimizer's update method is already called.